### PR TITLE
adding integ tests for S3Control to get better coverage for inconsist…

### PR DESCRIPTION
**Pull request by cenedhryn**
…ent exception unmarshalling

## Description
Adds integration tests for s3control

## Motivation and Context
There is an ongoing issue which is now partially fixed:
https://github.com/aws/aws-sdk-java-v2/issues/1986

These tests show that the error code is now correctly returned, but it does not return an InvalidRequestException, it's parsed as an S3ControlException, so the fix is not complete. 

## Testing
Local integ tests

### CODE REVIEW

File: services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+import org.assertj.core.api.ThrowableAssert;
```
Comment (Quanzzzz) [inline services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java:23]: Looks like this dependency wasn't used?

File: services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java [hunk 2]
```diff
@@ -29,6 +30,7 @@
 import software.amazon.awssdk.http.SdkHttpFullRequest;
 import software.amazon.awssdk.services.s3control.model.DeletePublicAccessBlockRequest;
 import software.amazon.awssdk.services.s3control.model.GetPublicAccessBlockResponse;
+import software.amazon.awssdk.services.s3control.model.InvalidRequestException;
```
Comment (Quanzzzz) [inline services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java:33]: Looks like this dependency also wasn't used?

File: services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java [hunk 3]
```diff
@@ -102,6 +104,40 @@ public void getPublicAccessBlock_NoSuchAccount() {
         }
     }
 
+    @Test
+    public void getPublicAccessBlock_NoSuchPublicAccessBlock() {
+        try {
+            client.getPublicAccessBlock(r -> r.accountId(accountId));
+            fail("Expected exception");
+        } catch (S3ControlException e) {
+            assertEquals("NoSuchPublicAccessBlockConfiguration", e.awsErrorDetails().errorCode());
```
Comment (Quanzzzz) [inline services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java:113]: Can we change this to `assertThat`? And same as all the other usages below? `assertThat` is more readable and flexible, and it's used more often in our code base.

File: services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java (hunk 1 above)
Comment (cenedhryn) [reply services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java:23]: Missed that, built with the quick flag...

File: services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java (hunk 3 above)
Comment (cenedhryn) [reply services/s3control/src/it/java/software.amazon.awssdk.services.s3control/S3ControlIntegrationTest.java]: I prefer assertThat too. Changed all assertEquals in the file (but did not touch the assertTrue etc)

**Review by Quanzzzz — APPROVED** [review]

**Pull request merged**